src/tainting/Taint_shape.ml

```diff
@@ -0,0 +1,302 @@
+(* Iago Abal
+ *
+ * Copyright (C) 2022-2024 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+open Common
+module T = Taint
+module Taints = T.Taint_set
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
```

Comment:
Please add a high level overview of the goal of the module here, just after the prelude.